### ISSUE TITLE
deserializers break when using with immutables

### ISSUE THREAD

**Issue by brentryan** [jonpeterson/jackson-module-model-versioning#6]
https://github.com/immutables/immutables/issues/461

@jonpeterson I'm trying to hunt this issue down and I'm stumped a bit.

Let me know if you have any ideas.

```
@Immutable
@JsonSerialize(as = ImmutableEntity.class)
@JsonDeserialize(as = ImmutableEntity.class)
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonVersionedModel(
        currentVersion = "1",
        defaultDeserializeToVersion = "0",
        versionToSuppressPropertySerialization = "0",
        toCurrentConverterClass = EntityCurrentConverter.class,
        toPastConverterClass = EntityPastConverter.class
)
@Value.Style(additionalJsonAnnotations = {JsonVersionedModel.class})
public interface Entity {

    int MIN_LEN_ANOTHER_FIELD = 2;
    int MAX_LEN_ANOTHER_FIELD = 10;

    @JsonSerializeToVersion(defaultToSource = true)
    @Nullable
    String getSerializeToVersion();

    /**
     * @return The field
     */
    String field();

    /**
     * @return another field
     */
    @Size(min = MIN_LEN_ANOTHER_FIELD, max = MAX_LEN_ANOTHER_FIELD)
    String getAnotherField();
}
```

Essentially using jackson to deserialize a string like:

```
        ObjectMapper m = new ObjectMapper();
        m.registerModule(new VersioningModule());

        String json
                = "{\"field\":\"f\",\"anotherField\":\"a\",\"modelVersion\":\"1\"}";

        Entity pe = m.readValue(json, Entity.class);
```

Winds up causing issues because VersionedModelDeserializer.deserialize() gets called twice and the modelVersion is removed in the 2nd pass making it seem as if the modelVersion should be "0" when it should be "1".

Thoughts?

**Comment by brentryan** [comment]
Found a way around this by writing a custom deserializer.  I'm going to put in a new PR which allows us to extend this module so it works with immutables without needing to change this project.  See https://github.com/immutables/immutables/issues/461 for more details.

**Comment by jonpeterson** [comment]
Merged and released to Maven Central as 1.2.2.  Thanks.
